[BUG] Template specification done twice?

**Describe the bug**
See #215 and #216 , somehow there are two function definitions compiled twice.

**To Reproduce**
Steps to reproduce the behavior:
1. Check out #215 or #216 .
2. Compile with 4x1.
3. Compile UnitTest->Schema->4x1->surface_model

**Expected behavior**
It compiles without linker error LNK2005.

**Screenshots**
See #215 or #216 .

**CMake configuration**
 If applicable, list any special configurations you have made in CMake.
- EARLYBINDING_WITH_IFC4x1: selected

**Additional context**
Probably, it is because of [this]/https://stackoverflow.com/questions/25529893/explicit-specialization-of-function-templates-causes-linker-error?rq=1).

--- a/Core/src/IfcGeometryConverter/PlacementConverterImpl.cpp
+++ b/Core/src/IfcGeometryConverter/PlacementConverterImpl.cpp
@@ -1,0 +1,61 @@
+/*
+    Copyright (c) 2020 Technical University of Munich
+    Chair of Computational Modeling and Simulation.
+
+    TUM Open Infra Platform is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License Version 3
+    as published by the Free Software Foundation.
+
+    TUM Open Infra Platform is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifdef OIP_MODULE_EARLYBINDING_IFC4X1
+#include "EMTIFC4X1EntityTypes.h"
+#include "IFC4X1Entities.h"
+#endif
+
+#include "PlacementConverter.h"
+#include "PlacementConverterImpl.h"
+
+namespace OpenInfraPlatform {
+    namespace Core {
+        namespace IfcGeometryConverter {
+
+			// IFC 4x1 specifics
+		#ifdef OIP_MODULE_EARLYBINDING_IFC4X1
+			template <>
+			EXPRESSReference<emt::IFC4X1EntityTypes::IfcBoundedCurve> PlacementConverterT<emt::IFC4X1EntityTypes>::getCurveOfPlacement(
+                const EXPRESSReference<emt::IFC4X1EntityTypes::IfcLinearPlacement>& linearPlacement
+			) const
+            {
+				// check input
+				if (linearPlacement.expired())
+					throw oip::ReferenceExpiredException(linearPlacement);
+				// check if correct type
+				if (!linearPlacement->PlacementRelTo.isOfType<emt::IFC4X1EntityTypes::IfcBoundedCurve>())
+					throw oip::InconsistentModellingException(linearPlacement, "Only a IfcBoundedCurve can be used as a reference curve.");
+				// return the curve
+                return linearPlacement->PlacementRelTo.as<emt::IFC4X1EntityTypes::IfcBoundedCurve>();
+            }
+			
+			template <>
+			double PlacementConverterT<emt::IFC4X1EntityTypes>::convertRelativePlacement(
+				const EXPRESSReference<emt::IFC4X1EntityTypes::IfcLinearPlacement>& linear_placement,
+				std::vector<EXPRESSReference<emt::IFC4X1EntityTypes::IfcObjectPlacement>>& alreadyApplied
+			) const
+            {
+				// no relative placement in IFC4x1
+                return 0.;
+            }
+		#endif
+
+        }
+    }
+}
+

--- a/Core/src/IfcGeometryConverter/PlacementConverterImpl.h
+++ b/Core/src/IfcGeometryConverter/PlacementConverterImpl.h
@@ -26,28 +26,16 @@ namespace OpenInfraPlatform {
 
 			// IFC 4x1 specifics
 		#ifdef OIP_MODULE_EARLYBINDING_IFC4X1
+			template <>
             EXPRESSReference<emt::IFC4X1EntityTypes::IfcBoundedCurve> PlacementConverterT<emt::IFC4X1EntityTypes>::getCurveOfPlacement(
                 const EXPRESSReference<emt::IFC4X1EntityTypes::IfcLinearPlacement>& linearPlacement
-			) const
-            {
-				// check input
-				if (linearPlacement.expired())
-					throw oip::ReferenceExpiredException(linearPlacement);
-				// check if correct type
-				if (!linearPlacement->PlacementRelTo.isOfType<emt::IFC4X1EntityTypes::IfcBoundedCurve>())
-					throw oip::InconsistentModellingException(linearPlacement, "Only a IfcBoundedCurve can be used as a reference curve.");
-				// return the curve
-                return linearPlacement->PlacementRelTo.as<emt::IFC4X1EntityTypes::IfcBoundedCurve>();
-            };
+			) const;
 			
-            double PlacementConverterT<emt::IFC4X1EntityTypes>::convertRelativePlacement(
+			template <>
+			double PlacementConverterT<emt::IFC4X1EntityTypes>::convertRelativePlacement(
 				const EXPRESSReference<emt::IFC4X1EntityTypes::IfcLinearPlacement>& linear_placement,
 				std::vector<EXPRESSReference<emt::IFC4X1EntityTypes::IfcObjectPlacement>>& alreadyApplied
-			) const
-            {
-				// no relative placement in IFC4x1
-                return 0.;
-            }
+			) const;
 		#endif
 
             /*! \brief Gets the \c IfcCurve attribute from \c IfcLinearPlacement.

